sandbox_dns_unix.go

```diff
@@ -275,7 +275,14 @@ func (sb *sandbox) updateDNS(ipv6Enabled bool) error {
 	if err != nil {
 		return err
 	}
-	if err = ioutil.WriteFile(tmpHashFile.Name(), []byte(newRC.Hash), filePerm); err != nil {
```

Comment:
These changes are a much needed improvement over what was there before.  Looking for these FD leaks and the code that caused them, its clear that their are some other issues and improvements that can be made.  These changes are small and use the APIs correctly without opening a file multiple times.  I think its worthwhile to take these changes as is because the previous usage and how the code is handling files are incorrect.